pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/elastic/migration/OsteoMigratorTest.java

```diff
@@ -0,0 +1,349 @@
+import org.broadinstitute.dsm.db.dao.tag.cohort.CohortTagDaoImpl;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.dto.ddp.participant.ParticipantDto;
+import org.broadinstitute.dsm.db.dto.tag.cohort.CohortTag;
+import org.broadinstitute.dsm.model.elastic.Dsm;
+import org.broadinstitute.dsm.model.elastic.search.ElasticSearchParticipantDto;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.CohortTagTestUtil;
+import org.broadinstitute.dsm.util.DdpInstanceGroupTestUtil;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.ElasticTestUtil;
+import org.broadinstitute.dsm.util.KitShippingTestUtil;
+import org.broadinstitute.dsm.util.MedicalRecordTestUtil;
+import org.broadinstitute.dsm.util.TestParticipantUtil;
+import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+@Slf4j
+public class OsteoMigratorTest extends DbAndElasticBaseTest {
+    private static final String TEST_USER = "TEST_USER";
+    private static final DDPInstanceDao ddpInstanceDao = new DDPInstanceDao();
+    private static final OncHistoryDetailDaoImpl oncHistoryDetailDao = new OncHistoryDetailDaoImpl();
+    private static DDPInstanceDto os1DdpInstanceDto;
+    private static DDPInstanceDto os2DdpInstanceDto;
+    private static String os1InstanceName;
+    private static String os2InstanceName;
+    private static String esIndex;
+    private static int participantCounter;
+    private static KitShippingTestUtil kitShippingTestUtil;
+    private static final List<ParticipantDto> os1Participants = new ArrayList<>();
+    private static final List<ParticipantDto> os2Participants = new ArrayList<>();
+    private static final Map<Integer, Integer> participantToMedicalRecordId = new HashMap<>();
+    private static final Map<Integer, Integer> participantToOncHistoryId = new HashMap<>();
+    private static final String OS1_TAG = "OS";
+    private static final String OS2_TAG = "OS PE-CGS";
+    private static final String baseInstanceName = "osteom";
+
+    private enum Cohort {
+        OS1, OS2, OS1_OS2
+    }
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        participantCounter = 1;
+        // NEW_OSTEO_INSTANCE_NAME and OLD_OSTEO_INSTANCE_NAME are already hardcoded in the cohort export code,
+        // so we either need to use them or change the code under test. Since the latter really needs a rewrite
+        // (with tests!) we do it this way for now.
+        // TODO: revisit this -- DC
+        os1InstanceName = OLD_OSTEO_INSTANCE_NAME;
+        os2InstanceName = NEW_OSTEO_INSTANCE_NAME;
+        esIndex = ElasticTestUtil.createIndex(baseInstanceName, "elastic/lmsMappings.json", null);
```

Comment:
Does it matter if we test this with OS mapping instead of LMS?

Reply:
I don't think so. At least for the things we are testing here.